modThread/modThread.py

```diff
@@ -16,268 +25,388 @@
-
-    async def _save_mods_category(self, ctx, mods_category):
-        await self.config.guild(ctx.guild).ModsCategory.set(mods_category)
-
-    async def _save_rules_role(self, ctx, rules_role):
-        await self.config.guild(ctx.guild).RulesRole.set(rules_role)
-
-    async def _rules_role(self, ctx):
-        return ctx.guild.get_role(await self.config.guild(ctx.guild).RulesRole())
+        syntax_desc="""
+```Syntax: mt groups add <group> <#category> <@role>
 
-    async def _save_numbers_role(self, ctx, numbers_role):
-        await self.config.guild(ctx.guild).NumbersRole.set(numbers_role)
+Example: ?mt groups add mods 1116910419458662490 @Mods```
+        """
+        groups_embed.add_field(
+            name="Syntax",
+            value=syntax_desc,
+            inline=False
+        )
 
-    async def _numbers_role(self, ctx):
-        return ctx.guild.get_role(await self.config.guild(ctx.guild).NumbersRole())
+        actions_desc = """
+- **add** - `?mt groups add <group> <#category> <@role>`
+- **update** - `?mt groups update <group> <#category> <@role>`
+- **delete** - `?mt groups delete <group>`
+        """
+        groups_embed.add_field(
+            name="Group Actions",
+            value=actions_desc,
+            inline=False
+        )
 
-    async def _save_mods_role(self, ctx, mods_role):
-        await self.config.guild(ctx.guild).ModsRole.set(mods_role)
+        groups_list = "- *None*"
+        if len(groups):
+            groups_list = ""
+            for group_name in groups:
+                group_obj = ctx.guild.get_role(groups[group_name]["role"])
+                groups_list += f"\n**{group_name}** - {group_obj.mention}"
+
+        groups_embed.add_field(
+            name="Defined Groups",
+            value=groups_list,
+            inline=False
+        )
 
-    async def _mods_role(self, ctx):
-        return ctx.guild.get_role(await self.config.guild(ctx.guild).ModsRole())
+        await ctx.send(embed=groups_embed)
+# endregion commands
+
+# region jsondb
+    async def _unset_group(
+        self,
+        guild: discord.Guild,
+        group_name: str | None,
+    ) -> NoReturn:
+        groups = await self._get_groups(guild)
+        groups.pop(group_name)
+
+        await self.config.guild(
+            guild
+        ).Groups.set(groups)
+
+    async def _set_group(
+        self,
+        guild: discord.Guild,
+        group_name: str | None,
+        category: discord.CategoryChannel | None,
+        role: discord.Role | None
+    ) -> NoReturn:
+        groups = await self._get_groups(guild)
+
+        group = {
+            "category": category.id,
+            "role": role.id
+        }
+
+        groups[group_name] = group
+        await self.config.guild(
+            guild
+        ).Groups.set(groups)
+
+    async def _get_primary_category(
+        self,
+        guild: discord.Guild
+    ) -> Optional[discord.CategoryChannel]:
+        return guild.get_channel(
```

Comment:
This should probably be replaced with

` return discord.utils.get(guild.categories, id=await self.config.guild(guild).PrimaryCategory())`

Reply:
Why? Any specific reason? I was following the previous code's conventions.

Reply:
Faster and more precise on type returns. Instead of getting it from a list of all the guilds "channels" which is a union of many types. We can simply query the categories within a guild which is a much smaller list, and guaranteed to return the expected type.

Reply:
Makes good sense to me. I'll make the change. Thanks!